File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/accounting/transactions/shared/mapping-detail.model.ts
export class MappingDetailModel{
    public AccountingMappingDetailId: number = 0;
    public GroupMappingId: number = 0; 
    public LedgerGroupId: number = 0;
    public DrCr: boolean = true;
    public LedgerReferenceId: number = 0;
    public Description: string = null;
    public LedgerGroupName: string = null; //not in database    
    public Name: string = "";
}

File: Code/Components/DanpheEMR.ServerModel/AccountingModels/Config/MappingDetailModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class MappingDetailModel
    {
        [Key]
        public int AccountingMappingDetailId  { get; set; }
        public int GroupMappingId { get; set; }

        /*Manipal-RevisionNeeded*/
        /*DevN:7Apr'23--We've made this Not NULL in Code, but not in Db yet.. This may impact in mutiple functions.. so recheck this again. */
        public int LedgerGroupId { get; set; }
        public bool DrCr { get; set; }
        public string Description { get; set; }
        [NotMapped]
        public string LedgerGroupName { get; set; }
        [NotMapped]
        public string Name { get; set; }
        [NotMapped]
        public int? LedgerReferenceId { get; set; }

    }
}
